### PR TITLE
Fix matplotlib backend warning when running system tests

### LINKED ISSUE #25884: EnginX System Test Warning
When running any system test, there is a warning message due to matplotlib importing in **EnginXScriptTest.py**.

System tests still run without issue.

Other people do not see this message when running system tests.

image

### Platforms affected
Ubuntu 18.04 at least
python 2 and python 3 release builds of Mantid

### PR DESCRIPTION
**Pull request by TTitcombe**
**Description of work.**
An import in a PyChop __init__ file was causing the matplotlib backend to be set.
The matplotlib backend is then set again when running system tests, which produces a warning.

Working with @martyngigg, we've commented out the import which was causing the issue

**To test:**
Run a system test. Before the test starts, you should see no matplotlib backend warnings.
Fixes #25884

*This does not require release notes* because **Internal change**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: scripts/PyChop/__init__.py [hunk 1]
```diff
@@ -26,10 +26,11 @@
...
 from .Instruments import Instrument as PyChop2  # noqa: F401
-# If the system doesn't have matplotlib, don't import the GUI.
-try:
-    from .PyChopGui import show as showGUI
-except ImportError:
-    def showGUI():
-        warnings.warn("PyChop GUI disabled: Cannot import Matplotlib.", RuntimeWarning)
-        return None
+
+# # If the system doesn't have matplotlib, don't import the GUI.
```
Comment (martyngigg) [inline scripts/PyChop/__init__.py]: Can we remove the commented out code?

File: qt/python/mantidqt/gui_helper.py [hunk 2]
```diff
@@ -18,6 +18,8 @@ def set_matplotlib_backend():
     This will set the backend if it hasn't been already. It also returns
     the name of the backend to be the name to be used for importing the
     correct matplotlib widgets.'''
+    if 'MPLBACKEND' in os.environ:
```
Comment (martyngigg) [inline qt/python/mantidqt/gui_helper.py]: Looking at this again perhaps we should move this under `backend = matplotlib.get_backend()` and then return backend? That way the caller can always guarantee a string return type?
Could we add a comment like "If the environment overrides the backend then we don't want to stamp on that" as a comment for the early exit?

Comment (TTitcombe) [reply qt/python/mantidqt/gui_helper.py]: As in 
```
if backend == "MPLBACKEND"
```
?

**Review by martyngigg — COMMENTED** [review]
This looks good to me.

@mducle Are you happy with this change? It allows us to import `PyChop` in the framework without  using any of the GUI layer which the system tests need to be able to do.

**Comment by mducle** [comment]
@martyngigg @TTitcombe Looks good to me. 

I forgot I'd put that check in - it now seems a bit uneccessary to me, but that was probably when the GUI was more integrated into the rest of the command.

File: qt/python/mantidqt/gui_helper.py [hunk 3]
```diff
@@ -19,6 +19,8 @@ def set_matplotlib_backend():
     the name of the backend to be the name to be used for importing the
     correct matplotlib widgets.'''
     backend = matplotlib.get_backend()
+    if 'MPLBACKEND' in os.environ:
```
Comment (martyngigg) [inline qt/python/mantidqt/gui_helper.py]: Looking at the test failures on Windows (I'm not sure why the other OSes have passed) then I think we will have to remove this `if`. The fix below will ensure the system tests are okay but `set_matplotlib_backend` is used in a way that means making the other tests work is a bit fiddly.

**Review by martyngigg — CHANGES_REQUESTED** [review]

**Comment by martyngigg** [comment]
@mducle Thanks for taking a look.

File: qt/python/mantidqt/gui_helper.py (hunk 3 above)
Comment (TTitcombe) [reply qt/python/mantidqt/gui_helper.py]: That's been removed. 

Removing the import in __init__ is sufficient to stop producing the warning

**Review by martyngigg — APPROVED** [review]
Thanks!

**Pull request merged**